src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomHeader/DatepickerCustomHeader.tsx

```diff
@@ -0,0 +1,141 @@
+import { CaretLeft, CaretRight, X } from '@phosphor-icons/react';
+import clsx from 'clsx';
+import format from 'date-fns/format';
+import getYear from 'date-fns/getYear';
+import isMatch from 'date-fns/isMatch';
+import parse from 'date-fns/parse';
+import React, {
+  type ChangeEventHandler,
+  type FC,
+  useEffect,
+  useState,
+} from 'react';
+
+import { formatText } from '~utils/intl.ts';
+import { range } from '~utils/lodash.ts';
+import InputBase from '~v5/common/Fields/InputBase/index.ts';
+
+import DatepickerYearDropdown from '../DatepickerYearDropdown/index.ts';
+
+import { type DatepickerCustomHeaderProps } from './types.ts';
+
+const years = range(1990, getYear(new Date()) + 1, 1);
```

Comment:
The min and max year values should be configurable via either props or constants _(same as `DEFAULT_DATE_FORMAT`)_ with a fallback to the values you provided above